lib/daemon_runner/client.rb

```diff
@@ -108,25 +126,60 @@ def parse_task(task)
     end
 
     # @private
+    # @param [Class] instance an instance of the task class
+    # @return [Hash<Symbol, String>] schedule parsed in parts: Schedule type and timing
+    def parse_schedule(instance)
+      valid_types = [:in, :at, :every, :interval, :cron]
+      out = {}
+      task_schedule = if instance.respond_to?(:schedule)
+        instance.send(:schedule)
+      else
+        schedule
+      end
+
+      raise ArgumentError, 'Not enough element in the Array' if task_schedule.length < 2
+      raise ArgumentError, 'Invalid schedule type' unless valid_types.include?(task_schedule[0].to_sym)
+
+      out[:type] = task_schedule[0].to_sym
+      out[:schedule] = task_schedule[1]
+      out
+    end
+
+    # @private
     # @param [Array<String, String, Array>] task to run
     # @return [String] output returned from task
     def run_task(task)
       parsed_task = parse_task(task)
       instance = parsed_task[:instance]
+      schedule = parse_schedule(instance)
       class_name = parsed_task[:class_name]
       method = parsed_task[:method]
       args = parsed_task[:args]
       log_line = "Running #{class_name}.#{method}"
       log_line += "(#{args})" unless args.empty?
       logger.debug log_line
 
-      out = if args.empty?
-              instance.send(method.to_sym)
-            else
-              instance.send(method.to_sym, args)
-            end
-      logger.debug "Got: #{out}"
-      out
+      # Schedule the task
+      schedule_log_line = "Scheduling job type #{schedule[:type]}"
+      schedule_log_line += " with interval #{schedule[:schedule]}"
+      logger.debug schedule_log_line
+      scheduler.send(schedule[:type], schedule[:schedule], :overlap => false, :job => true) do |job|
+        job[:error_sleep_time] = error_sleep_time
+        job[:logger] = logger
+
+        out = if args.empty?
+          instance.send(method.to_sym)
+        else
+          instance.send(method.to_sym, args)
+        end
+        logger.debug "Got: #{out}"
+      end
+    end
+
+    # @return [Rufus::Scheduler] A scheduler instance
+    def scheduler
+      return @scheduler unless @scheduler.nil?
+      @scheduler = ::Rufus::Scheduler.new
```

Comment:
I was following the convention @athompson-r7 used here: https://github.com/rapid7/daemon_runner/blob/master/lib/daemon_runner/client.rb#L38-L43